lib/googlestaticmap.rb

```diff
@@ -117,7 +117,7 @@ def url(protocol='http')
     attrs << ["key", @api_key] if !@api_key.nil?
     attrs << ["client", @client_id] if @api_key.nil? && !@client_id.nil? && !@private_key.nil?
     path << attrs.sort_by {|k,v| k}.collect {|attr| "#{attr[0]}=#{attr[1]}"}.join("&")
-    if @api_key.nil? && !@client_id.nil? && !@private_key.nil?
+    if (!@api_key.nil? ^ !@client_id.nil?) && !@private_key.nil?
```

Comment:
Any reason to use the exclusive or operator versus regular or (||)?  Seems like this would not apply the signature if both api_key and client_id are specified. Which isn't a real use case but in case the user does that, it should still append the signature.